processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -565,6 +566,32 @@ public Integer getDimensionIndex(String dimension)
     return dimensionOrder.get(dimension);
   }
 
+  public LinkedHashMap<String, Integer> getDimensionOrder()
+  {
+    return dimensionOrder;
+  }
+
+  public void loadDimensionOrder(LinkedHashMap<String, Integer> oldDimensionOrder)
```

Comment:
@nishantmonu51 I intended for it to be callable at any point; I just updated this code so it acquires a lock on dimensionOrdering before adding anything